include/boost/geometry/algorithms/detail/relate/turns.hpp

```diff
@@ -40,21 +40,20 @@ struct assign_policy
     static bool const include_degenerate = IncludeDegenerate;
 };
 
-// GET_TURNS
+// turn retriever, calling get_turns
 
 template
 <
     typename Geometry1,
     typename Geometry2,
+    typename Point,
     typename GetTurnPolicy = detail::get_turns::get_turn_info_type
         <
             Geometry1, Geometry2, assign_policy<>
         >
 >
-struct get_turns
+struct turn_retriever
```

Comment:
Why was this name changed? What was the original problem? That it's confused with the original `get_turns`?

Anyway this is semantically not a class which object is created somewhere like a functor/predicate whatever (it's not a retriever created and used somewhere) but implementation of an algorithm with static member function `apply()`. We use verbs for these in the library right?

BTW, if I was implementing this now it'd be free function `detail::relate::get_turns()`.

Reply:
See the PR's description: `get_turns` is a free function which does something else.

Reply:
Yes, `geometry::get_turns()` is a free function, `dispatch::get_turns` is a struct and `detail::get_turns_generic` is a struct as well (the implementation). `detail::relate::turns::get_turns` is an implementation as well. The only difference between `geometry::get_turns()` and `relate::turns::get_turns::apply()` is that parameters common across all relops and L/L setops are set here internally so the caller is not forced to pass them each time the same way. As I mentioned it could as well be implemented as a free function. It's neither a retriever nor a getter it's an implementation of an algorithm `get_turns()` specific for relops.

Reply:
I will also rename this back.